frontend/lib/map/location_button.dart

```diff
@@ -23,31 +26,37 @@ class _LocationButtonState extends State<LocationButton> {
   @override
   Widget build(BuildContext context) {
     var theme = Theme.of(context);
-    return Align(
-        alignment: Alignment.bottomRight,
-        child: Padding(
-            padding: const EdgeInsets.all(10),
-            child: FloatingActionButton(
-              heroTag: "fab_map_view",
-              elevation: 1,
-              backgroundColor: theme.backgroundColor,
-              child: AnimatedSwitcher(
-                child: _status == LocationPermissionStatus.requestFinished
-                    ? Icon(Icons.my_location,
-                        color: theme.colorScheme.secondary)
-                    : const SmallButtonSpinner(),
-                duration: const Duration(milliseconds: 200),
-              ),
-              onPressed: _status == LocationPermissionStatus.requestFinished
-                  ? _onPressed
-                  : null,
-            )));
+    final settings = Provider.of<SettingsModel>(context);
+    if (settings.locationFeatureEnabled) {
```

Comment:
In #408 we show the FAB even though the feature is disabled. This provides a way of reenabling it.